### ISSUE TITLE
When I trained with two processes the gradient of the parameters could not be shared and I ended up with two different models. How to solve this problem?

### ISSUE THREAD

**Issue by zypsjtu** [huggingface/accelerate#2262]
When I trained with two processes the gradient of the parameters could not be shared and I ended up with two different models. Did anyone meet this problem before? How to solve it?

**Comment by muellerzr** [comment]
Yes, after the `.backward()` these are averaged in DDP and you will be left with the same model in the end.

**Comment by zypsjtu** [comment]
> Yes, after the `.backward()` these are averaged in DDP and you will be left with the same model in the end.

thanks for your reply.
i printed the parameters of my model and got different results in two devices. 
```code
                accelerator.backward(loss)
                print((hypernetwork.module.weights[0]))
```
and after training, when i try to save my model, i can just save the model in the device 0.
```code
   if accelerator.is_main_process:
        ......
        torch.save(state_dict, os.path.join(args.output_dir, "optimized.bin"))
```

**Comment by muellerzr** [comment]
Can you please give us a full minimal reproducer? And the output of `accelerate env`

**Comment by zypsjtu** [comment]
```
- `Accelerate` version: 0.21.0
- Platform: Linux-6.2.0-33-generic-x86_64-with-glibc2.17
- Python version: 3.8.10
- Numpy version: 1.23.5
- PyTorch version (GPU?): 2.0.1 (True)
- PyTorch XPU available: False
- PyTorch NPU available: False
- System RAM: 62.61 GB
- GPU type: NVIDIA GeForce RTX 4090
- `Accelerate` default config:
        - compute_environment: LOCAL_MACHINE
        - distributed_type: MULTI_GPU
        - mixed_precision: bf16
        - use_cpu: False
        - num_processes: 2
        - machine_rank: 0
        - num_machines: 1
        - gpu_ids: all
        - rdzv_backend: static
        - same_network: True
        - main_training_function: main
        - downcast_bf16: no
        - tpu_use_cluster: False
        - tpu_use_sudo: False
        - tpu_env: []
```

i found that in DDP there is a command ```find_unused_parameters = True``` which is similar to my problem. How do I use that in Accelerate?

**Comment by SunMarc** [comment]
Hi @zypsjtu, you can pass additional kwargs this way: 
```python
from accelerate import DistributedDataParallelKwargs

ddp_kwargs = DistributedDataParallelKwargs(find_unused_parameters=True)
accelerator = Accelerator(kwargs_handlers=[ddp_kwargs])
```

**Comment by liangyuxin42** [comment]
May I ask if you have solved this problem? I'm in a similar situation and would like a little direction, thanks!
